docs/configuration.asciidoc

```diff
@@ -0,0 +1,131 @@
+////
+Goal of this doc:
+Provide a complete reference of all available configuration options and where/how they can be set. (Any Elastic-specific configuration options are listed directly. General OpenTelemetry configuration options are linked.)
+
+Assumptions we're comfortable making about the reader:
+* They are familiar with Elastic
+* They are familiar with OpenTelemetry
+////
+
+[[configuration]]
+== Configure
+
+:language: Java
+:language_lc: java
+:distro_name: Elastic Distribution for OpenTelemetry {language}
+
+include::release-status.asciidoc[]
+
+////
+TO DO: How do users set configuration options?
+////
+[discrete]
+[[configure-methods]]
+=== Configuration methods
+
+////
+TO DO:
+Are there multiple ways to set configuration options?
+For example, can they use environment variables OR a config file?
+If there are, what are they and why would you use each method?
+Where/how do you use each method?
+Provide an example to illustrate the syntax.
+////
+
+////
+TO DO:
+If configuration options are defined using multiple methods,
+which one takes precedence over the other(s)?
+////
+
+[discrete]
+[[configuration-options]]
+=== Configuration options
+
+Because the Elastic Distribution for OpenTelemetry Java ("the distro") is an extension of the https://github.com/open-telemetry/opentelemetry-java-instrumentation[OpenTelemetry Java agent], it supports both:
+
+* General OpenTelemetry SDK configuration options
+* Elastic-specific configuration options that are only available when using the distro
+
+[discrete]
+[[configure-otel-sdk-options]]
+==== OpenTelemetry SDK configuration options
+
+////
+TO DO:
+Is this true?
+Are there any options that aren't supported?
+Any options that shouldn't be used?
+////
+The distro supports all configuration options listed in the https://opentelemetry.io/docs/languages/sdk-configuration/general/[OpenTelemetry General SDK Configuration documentation].
+
+////
+TO DO:
+Are there Elastic-specific custom configuration options
+in addition to the general OpenTelemetry SDK
+configuration options? Or are all the customizations
+"behind the scenes"?
+If not, you can delete this section.
+////
+[discrete]
+[[configure-distro-options]]
+==== Elastic-specific configuration options
```

Comment:
Linking to this page might be helpful: https://www.elastic.co/guide/en/observability/current/apm-secure-agent-communication.html

Reply:
I'm also realizing that we'll need to update pages like this in the APM Guide. For example, on this page we should add links to the otel books: https://www.elastic.co/guide/en/observability/current/apm-secret-token.html

Reply:
Opened https://github.com/elastic/observability-docs/issues/4089 to track